bazel/envoy_mobile_repositories.bzl

```diff
@@ -0,0 +1,127 @@
+    # supports platform toolchains.
+    http_archive(
+        name = "com_google_absl",
+        patches = ["//bazel:abseil.patch"],
+        sha256 = "14ee08e2089c2a9b6bf27e1d10abc5629c69c4d0bab4b78ec5b65a29ea1c2af7",
+        strip_prefix = "abseil-cpp-cf3a1998e9d41709d4141e2f13375993cba1130e",
+        # 2020-03-05
+        urls = ["https://github.com/abseil/abseil-cpp/archive/cf3a1998e9d41709d4141e2f13375993cba1130e.tar.gz"],
+    )
+
+    # This should be kept in sync with Envoy itself, we just need to apply this patch
+    # Remove this once https://boringssl-review.googlesource.com/c/boringssl/+/37804 is in master-with-bazel
+    http_archive(
+        name = "boringssl",
+        patches = ["//bazel:boringssl.patch"],
+        sha256 = "36049e6cd09b353c83878cae0dd84e8b603ba1a40dcd74e44ebad101fc5c672d",
+        strip_prefix = "boringssl-37b57ed537987f1b4c60c60fa1aba20f3a0f6d26",
+        urls = ["https://github.com/google/boringssl/archive/37b57ed537987f1b4c60c60fa1aba20f3a0f6d26.tar.gz"],
+    )
+
+def swift_repos():
+    http_file(
+        name = "xctestrunner",
+        executable = 1,
+        sha256 = "0338c71977106f1304a8056739db6f462a76f386a299052c1ed7f8fd463d01a8",
+        urls = ["https://github.com/google/xctestrunner/releases/download/0.2.11/ios_test_runner.par"],
+    )
+
+    http_archive(
+        name = "build_bazel_rules_apple",
+        sha256 = "ee9e6073aeb5a65c100cb9c44b0017c937706a4ae03176e14a7e78620a198079",
+        strip_prefix = "rules_apple-5131f3d46794bf227d296c82f30c2499c9de3c5b",
+        url = "https://github.com/bazelbuild/rules_apple/archive/5131f3d46794bf227d296c82f30c2499c9de3c5b.tar.gz",
+    )
+
+    http_archive(
+        name = "build_bazel_rules_swift",
+        sha256 = "d0833bc6dad817a367936a5f902a0c11318160b5e80a20ece35fb85a5675c886",
+        strip_prefix = "rules_swift-3eeeb53cebda55b349d64c9fc144e18c5f7c0eb8",
+        url = "https://github.com/bazelbuild/rules_swift/archive/3eeeb53cebda55b349d64c9fc144e18c5f7c0eb8.tar.gz",
+    )
+
+def kotlin_repos():
+    http_archive(
+        name = "rules_jvm_external",
+        sha256 = "1bbf2e48d07686707dd85357e9a94da775e1dbd7c464272b3664283c9c716d26",
+        strip_prefix = "rules_jvm_external-2.10",
+        url = "https://github.com/bazelbuild/rules_jvm_external/archive/2.10.zip",
+    )
+
+    rules_kotlin_version = "legacy-1.3.0-rc2"
+
+    rules_kotlin_sha = "dc1c76f91228ddaf4f7ca4190b82d61939e95369f61dea715e8be28792072b1b"
+
+    http_archive(
+        name = "io_bazel_rules_kotlin",
+        sha256 = rules_kotlin_sha,
+        strip_prefix = "rules_kotlin-%s" % rules_kotlin_version,
+        type = "zip",
+        urls = ["https://github.com/bazelbuild/rules_kotlin/archive/%s.zip" % rules_kotlin_version],
+    )
+
+    rules_detekt_version = "0.3.0"
+
+    rules_detekt_sha = "b1b4c8a3228f880a169ab60a817619bc4cf254443196e7e108ece411cb9c580e"
+
+    http_archive(
```

Comment:
I am trying to understand the difference between the kotlin_repos and the the kotlin_dependencies (same for swift)

Reply:
There's a few concepts here: `http_archive`, `load()`, and actually calling the initialization method.

The `http_archive` is needed to download the remote repository, to call the initialization methods in the repository, we have to call the `load()` function to import the initialization method.

Here's the nuance of a `WORKSPACE` and a `bzl` file:
1. `load()` methods have to be declared at the very top of every `bzl` file
2. Some `load()` methods depend on other repositories which means we need to download these repositories first before we call `load()`
3. `load()` cannot be called within a `bzl` method so when we load a specific `bzl` file, we essentially call all the `load()` methods defined for the `bzl` file

This means that in the `WORKSPACE` we have more freedom to define things in a specific fragile order but when we move these to a `bzl` file, it requires some thoughtful organization (I did my best!). It's also the reason we have so many different files to replace whatever we had in `WORKSPACE`

Reply:
> This means that in the `WORKSPACE` we have more freedom to define things in a specific fragile order but when we move these to a `bzl` file, it requires some thoughtful organization

Ah this is the key point. I see! Thanks